### DISCUSSION TITLE
How to download Linux AND Windows engines on a Linux CI sytem?

### DISCUSSION THREAD

**Discussion by tmtron** [prisma/prisma#14092] [Q&A] [answered]
What we want to achieve is to build linux and windows installers for our Electron app on our (Linux) build server.

After installing Wine, this basically works (i.e. we can build the Installers), except for Prisma Migration:
i.e. Prisma relies on OS-specific binaries - thus, we must have the correct binaries available for each platform and then include them in the Electron installer

## Prisma Query engine / Prisma Client 
This part works: we set the [binaryTarget][1] to our target platforms: `binaryTargets = ["windows", "debian-openssl-1.1.x"]`.  
When we now generate the client (`prisma generate`), the target dir (`node_modules/.prisma/client/`) contains the query-engines for linux and windows: `libquery_engine-debian-openssl-1.1.x.so.node`, `query_engine-windows.dll.node`

## Prisma Migration Engine
i.e. in `package.json` we only specify the `prisma` dependency and npm will only install the binaries for the current OS in `node_modules/@prisma/engines/`, i.e.: `libquery_engine-debian-openssl-1.1.x.so.node`, `migration-engine-debian-openssl-1.1.x`

How can we get both migration Engines (i.e. for Windows and Linux)?
* i.e. maybe there is some config that I've missed, like [binaryTarget][1] for the Prisma Client?
* or maybe we can tell npm to download additional files?

Reference: [Stackoverflow Question](https://stackoverflow.com/questions/72784805/how-to-build-prisma-windows-electron-app-on-linux)

  [1]: https://www.prisma.io/docs/concepts/components/prisma-schema/generators#the-native-binary-target

**Comment by janpio** [comment, marked as answer]
I think you are looking for https://www.prisma.io/docs/reference/api-reference/environment-variables-reference#cli-binary-targets

**Reply by tmtron** [reply]
Thanks, exactly what I was looking for.

Notes:
* for some reason `npm install` did not download the binaries for other platforms after setting the env-var `PRISMA_CLI_BINARY_TARGETS`
* I had to 
  * delete `node_modules/@prisma/engines`
  * then `npm install` again 
  * and now all expected binaries exist in `node_modules/@prisma/engines`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
